nodeadm/internal/kubelet/image-credential-provider.go

```diff
@@ -74,6 +88,8 @@ func generateImageCredentialProviderConfig(cfg *api.NodeConfig, ecrCredentialPro
 	} else {
 		templateVars.ConfigApiVersion = "kubelet.config.k8s.io/v1"
 		templateVars.ProviderApiVersion = "credentialprovider.kubelet.k8s.io/v1"
+		// ecr-credential-provider has support for public.ecr.aws in 1.27+
+		templateVars.MatchImages = append(templateVars.MatchImages, "public.ecr.aws")
```

Comment:
ignore me -- I breezed past this part.

Nick's suggestion to use a go type is probably reasonable given how clunky loops are in a go template. Looks like the versions we need are still available in the latest k/k: `k8s.io/kubelet/config/v1alpha1`, etc